src/tables.jl

```diff
@@ -47,3 +47,585 @@ function schema_names(featuredefn::IFeatureDefnView)
+                            pop!(maybegeomtypes, j)
+                        end
+                    end
+                end
+                maybeWKBcolinds = maybeWKBcolinds ∩ keys(maybegeomtypes)
+                testWKB = !Base.isempty(maybeWKBcolinds)
+            end
+            state = iterate(rows, st)
+            state === nothing && break
+            row, st = state
+        end
+        state === nothing && begin
+            WKxgeomcolinds = findall(T -> T != Union{}, maybegeomtypes)
+            for j in WKxgeomcolinds
+                AGtypes[j] = (
+                    _convert_cleantype_to_AGtype ∘
+                    _convert_coltype_to_cleantype
+                )(
+                    maybegeomtypes[j],
+                )
+            end
+        end
+    end
+
+    # Verify after parsing that:
+    # - there is no column, not specified in `geomcols` kwarg, and found to be 
+    #   of a geometry eltype which is not a compatible GDAL field type 
+    #   (e.g. `IGeometry` or `GeoInterface.AbstractGeometry`) 
+    # - there is no column specified in `geomcols` kwarg that could not be
+    #   parsed as a geometry column
+    if spgeomcols !== nothing
+        foundgeomcols = findall(T -> T isa OGRwkbGeometryType, AGtypes)
+        if Set(spgeomcols) != Set(foundgeomcols)
+            diff = setdiff(spgeomcols, foundgeomcols)
+            if !Base.isempty(diff)
+                error(
+                    "Column(s) $(join(string.(diff), ", ", " and ")) could not be parsed as geometry column(s)",
+                )
+            end
+            diff = setdiff(foundgeomcols, spgeomcols)
+            if !Base.isempty(diff)
+                error(
+                    "Column(s) $(join(string.(diff), ", ", " and ")) is(are) composed of geometry objects that cannot be converted to a GDAL field type.\nConsider adding this(these) column(s) to `geomcols` kwarg or convert their values to WKT/WKB",
+                )
+            end
+        end
+    end
+
+    return AGtypes
+end
+
+"""
+    _coherencecheckandnormalizationofkwargs(geomcols, fieldtypes)
+
+Test coherence: 
+   - of `geomcols` and `fieldtypes` kwargs with table schema
+   - between `geomcols` and `fieldtypes` kwargs
+   - of `ORGFieldTypes` and `OGRFieldSubType` types in `fieldtypes`kwarg
+
+And normalize `geomcols` and `fieldtypes` kwargs with indices of table schema names.
+
+"""
+function _coherencecheckandnormalizationofkwargs(
+    geomcols::Union{Nothing,Vector{String},Vector{Int}},
+    fieldtypes::Union{
+        Nothing,
+        Dict{Int,Union{OGRwkbGeometryType,Tuple{OGRFieldType,OGRFieldSubType}}},
+        Dict{
+            String,
+            Union{OGRwkbGeometryType,Tuple{OGRFieldType,OGRFieldSubType}},
+        },
+    },
+    colnames = Vector{String},
+)::Tuple{
+    Union{Nothing,Vector{Int}},
+    Union{
+        Nothing,
+        Dict{Int,Union{OGRwkbGeometryType,Tuple{OGRFieldType,OGRFieldSubType}}},
+    },
+}
```

Comment:
To the extent I can, I'd prefer to avoid creating/supporting an API that requires such a function. (Nonetheless, I want to say I really really appreciate the work that went into this function.)

Reply:
That was what I was fearing when I asked:

> But is it worth the extra complexity ?

in https://github.com/yeesian/ArchGDAL.jl/pull/243#discussion_r738021766 :-)

But now, it is done, I think it will be helpful for most, and I will help you maintain it. We could just drop `fieldtypes` kwarg for which personally I see no usage.

Reply:
I agree it will be helpful for most, and we can drop the `fieldtypes` kwarg, thank you for demonstrating it